### PR TITLE
docs: replace Bazel and Intellij documentation with SBT and VSCode (DEV-607)

### PR DESCRIPTION
**Pull request by BalduinLandolt**
Resolves DEV-607

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by irinaschubert — APPROVED** [review]
LGTM, just minor stuff

File: docs/05-internals/development/third-party.md [hunk 1]
```diff
@@ -1,16 +1,51 @@
...
+### Referencing a third party library
+
+There is an object `Dependencies` where each library should be declared in a `val`.
+
+```scala
+val akkaHttpCors = "ch.megard" %% "akka-http-cors" % "1.0.0"
+```
+
+The first string corresponds to the group/organization in the library's maven artefact,
+the second sting corresponds to the artefact ID and the third string defines the version.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/third-party.md:19]: done

File: docs/05-internals/development/vscode-config.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Setup Visual Studio Code for development of DSP-API
+
+To have full functionality, the "Scala (Metals)" plugin should be installed.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/vscode-config.md:3]: I generally didn't add too many links all over the place. Do you think I should? (So far, this seems to be pretty common in the docs, but I feel a bit silly for example to add a link to each instance of "VSCode" in the text...)

File: docs/05-internals/development/vscode-config.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+As a formatter, we use [Scalafmt](https://scalameta.org/scalafmt/).
+Metals automatically recognizes the formatting configuration in the `.scalafmt.conf` file in the root directory.
+VSCode should be configured so that it austomatically formats (e.g. on file saved).
+
+
+## Running Tests
+
+The tests can be run through make commands or through SBT.
+The most convenient way to run the tests though, is through VSCode.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/vscode-config.md:23]: done

File: docs/05-internals/development/vscode-config.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Or with the setting `"metals.testUserInterface": "Code Lenses"` directly in the text:
+
+![Tests in VSCode with Codelens Enabled](figures/vscode-metals-test-codelens.png)
+
+
+## Debugger
+
+It is currently not possible to start the stack in debug mode.
+
+Tests can be run in debug mode by running them as described above but chosing `debug test` instead of `test`.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/vscode-config.md:37]: done, thanks

File: mkdocs.yml [hunk 5]
```diff
@@ -6,147 +6,143 @@ edit_uri: blob/main/docs/
...
+                      - Setup Visual Studio Code for development of DSP-API: 05-internals/development/vscode-config.md
+                      - Testing: 05-internals/development/testing.md
+                      - Docker Cheat Sheet: 05-internals/development/docker-cheat-sheet.md
+                      - Monitoring DSP-API: 05-internals/development/monitoring.md
+                      - Starting the DSP-Stack inside Docker Container: 05-internals/development/docker-compose.md
+                      - Updating Repositories: 05-internals/development/updating-repositories.md
+                      - Generating Client Test Data: 05-internals/development/generating-client-test-data.md
+                      - Third-Party Dependencies: 05-internals/development/third-party.md
+      - SIPI:
+              - Index: 07-sipi/index.md
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply mkdocs.yml:119]: I totally overlooked this... probably should be adjusted, right? What do the others think?

File: docs/05-internals/development/vscode-config.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# Setup Visual Studio Code for development of DSP-API
+
+To have full functionality, the [Scala Metals](https://scalameta.org/metals/) plugin should be installed.
+
+Additionally, a number of plugins can be installed for convenience, but are not required. Those include but are by no means limited to:
+- Docker (to attach to running docker containers)
+- Stardog RDF grammar (.ttl sysntax highlighting)
+- Lua
+- REST client
```
Comment (mpro7) [inline docs/05-internals/development/vscode-config.md:9]:
```suggestion
- Docker - to attach to running docker containers,
- Stardog RDF grammar - TTL syntax highlighting,
- Lua,
- REST client,
```

File: mkdocs.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (mpro7) [reply mkdocs.yml:119]: IMO it's pretty obvious the numbering needs to be continued.

**Review by mpro7 — CHANGES_REQUESTED** [review]
Please have a look to my suggestions.

File: README.md [hunk 7]
```diff
@@ -1,17 +1,17 @@
-# Knora &mdash; Knowledge Organization, Representation, and Annotation
+# DSP-API &mdash; DaSCH Service Platform API
 
 [![Github](https://img.shields.io/github/v/tag/dasch-swiss/dsp-api?include_prereleases&label=Github%20tag)](https://github.com/dasch-swiss/dsp-api)
 [![Docker](https://img.shields.io/docker/v/daschswiss/knora-api?label=Docker%20image)](https://hub.docker.com/r/daschswiss/knora-api)
 [![CI](https://github.com/dasch-swiss/dsp-app/workflows/CI/badge.svg)](https://github.com/dasch-swiss/dsp-api/actions?query=workflow%3ACI)
 
-[Knora](https://www.knora.org/) is a server application for storing, sharing, and working with primary sources and data in the humanities.
+[DSP API](https://admin.dasch.swiss/) is a server application for storing, sharing, and working with primary sources and data in the humanities.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply README.md:7]: You're right

File: docs/05-internals/development/third-party.md [hunk 8]
```diff
@@ -1,16 +1,51 @@
...
+There is an object `Dependencies` where each library should be declared in a `val`.
+
+```scala
+val akkaHttpCors = "ch.megard" %% "akka-http-cors" % "1.0.0"
+```
+
+The first string corresponds to the group/organization in the library's maven artefact,
+the second string corresponds to the artefact ID and the third string defines the version.
+
+The strings are combined with the `%`-operator or the `%%`-operator, the latter fixing the dependency to the specified scala-version.
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/third-party.md:21]: done

File: docs/05-internals/development/third-party.md [hunk 9]
```diff
@@ -1,16 +1,51 @@
...
-The following section discusses on how third-party dependencies should be defined.
+Third party libraries are managed by SBT.
 
-Third-party dependencies are defined in the `third_party` folder. There are
-two main types of dependencies: (1) Maven library dependencies and (2) Docker
-image versions.
+## Defining Dependencies in `Dependencies.scala`
+
+Within the `build.sbt` file, the `Dependencies` package is referenced, which is located in `project/Dependencies.scala`.
+All third party dependencies need to be declared there:
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply docs/05-internals/development/third-party.md:8]: Done

**Review by subotic — APPROVED** [review]

File: docs/05-internals/development/vscode-config.md (hunk 6 above)
Comment (BalduinLandolt) [reply docs/05-internals/development/vscode-config.md:9]: I left out the comma at the end of the bulletpoint entries, but apart from that, I applied the suggestion

**Review by mpro7 — APPROVED** [review]

**Pull request merged**